src/Protocols/NFS/nfs3_read.c

```diff
@@ -148,8 +148,7 @@ static int nfs3_complete_read(struct nfs3_read_data *data)
 
 static enum xprt_stat nfs3_read_resume(struct svc_req *req)
 {
-	SVCXPRT *xprt = req->rq_xprt;
-	nfs_request_t *reqdata = xprt->xp_u1;
+	nfs_request_t *reqdata = container_of(req, nfs_request_t, svc);
```

Comment:
oops, yeah, we should no longer rely on xprt->xp_u1 directly, I suspect

Reply:
From what I see it is always null; not certain when it broke but I might be the first using the full async implementation and therefore the *_resume methods.

This at least allowed me to perform a single nfs v3 write. In a heavier workload, I'm crashing at nfs3_write.c line 152 with null `proc_data` Still investigating that one.

Reply:
NFS4 does store reqdata in xp_u1.  Presumably not necessary, since the container_of form should work.  It might be better to not depend on xp_u1.